src/graph/TraverseExecutor.cpp

```diff
@@ -138,10 +149,132 @@ VariantType Collector::getProp(const std::string &prop,
+    DCHECK(outputSchema != nullptr);
+    DCHECK_EQ(vals.size(), colNames.size());
+    DCHECK_EQ(vals.size(), colTypes.size());
+    using nebula::cpp2::SupportedType;
+    auto index = 0u;
+    for (auto &it : colTypes) {
+        SupportedType type;
+        if (it == SupportedType::UNKNOWN) {
+            switch (vals[index].which()) {
+                case VAR_INT64:
+                    // all integers in InterimResult are regarded as type of INT
+                    type = SupportedType::INT;
+                    break;
+                case VAR_DOUBLE:
+                    type = SupportedType::DOUBLE;
+                    break;
+                case VAR_BOOL:
+                    type = SupportedType::BOOL;
+                    break;
+                case VAR_STR:
+                    type = SupportedType::STRING;
+                    break;
+                default:
+                    std::string errMsg =
+                        folly::stringPrintf("Unknown VariantType: %d", vals[index].which());
+                    LOG(ERROR) << errMsg;
+                    return Status::Error(errMsg);
+            }
+        } else {
+            type = it;
+        }
+
+        outputSchema->appendCol(colNames[index], type);
+        index++;
+    }
+    return Status::OK();
+}
+
+Status YieldClauseWrapper::prepare(
+        const InterimResult *inputs,
+        const VariableHolder *varHolder,
+        std::vector<YieldColumn*> &yields) {
+    auto cols = clause_->columns();
+    yieldColsHolder_ = std::make_unique<YieldColumns>();
+    for (auto *col : cols) {
+        if (col->expr()->isInputExpression()) {
+            if (needAllPropsFromInput(col, inputs, yields)) {
+                continue;
+            }
+        } else if (col->expr()->isVariableExpression()) {
+            auto ret = needAllPropsFromVar(col, varHolder, yields);
+            if (!ret.ok()) {
+                return std::move(ret).status();
+            }
+            if (ret.value()) {
+                continue;
+            }
+        }
+        yields.emplace_back(col);
     }
+
+    return Status::OK();
 }
 
+bool YieldClauseWrapper::needAllPropsFromInput(const YieldColumn *col,
+                                               const InterimResult *inputs,
+                                               std::vector<YieldColumn*> &yields) {
+    auto *inputExpr = static_cast<InputPropertyExpression*>(col->expr());
+    auto *colName = inputExpr->prop();
+    if (*colName == "*") {
+        if (inputs != nullptr) {
+            auto schema = inputs->schema();
+            auto iter = schema->begin();
+            while (iter) {
+                auto *prop = iter->getName();
+                Expression *expr = new InputPropertyExpression(new std::string(prop));
+                YieldColumn *column = new YieldColumn(expr);
+                yieldColsHolder_->addColumn(column);
+                yields.emplace_back(column);
```

Comment:
Will it be assumed here that yieldColsHolder_ will definitely be released after yields?